purpleair/channel.py

```diff
@@ -175,22 +194,43 @@ def get_historical(self,
 
         columns = parent_cols if self.type == 'parent' else child_cols
         to_week = start_date - timedelta(weeks=1)
-        # pylint: disable=line-too-long
-        url = f'https://thingspeak.com/channels/{channel}/feed.csv?api_key={key}&offset=0&average=&round=2&start={to_week.strftime("%Y-%m-%d")}%2000:00:00&end={start_date.strftime("%Y-%m-%d")}%2000:00:00'
-        weekly_data = pd.read_csv(url)
-        if weeks_to_get > 1:
+        
+        # copy args to a local variable
+        thingspeak_args = thingspeak_args.copy()
+        default_args = {
+            'api_key': key,
+            'offset': 0,
+            'average': '',
+            'round': 2,
```

Comment:
This parameter should be configurable

Reply:
This is configurable if you pass thingspeak_args= {'round': 999} to the method call.

All these vals are taken from the args in the original URL that was in this repo. I'm happy to remove all of them though, if that's the preferred option.